databricks/koalas/frame.py

```diff
@@ -2897,6 +2897,157 @@ def merge(left, right: 'DataFrame', how: str = 'inner',
+
+        See Also
+        --------
+        DataFrame.merge: For column(s)-on-columns(s) operations.
+
+        Notes
+        -----
+        Parameters on, lsuffix, and rsuffix are not supported when passing a list of DataFrame
+        objects.
+
+        Examples
+        --------
+        >>> kdf1 = ks.DataFrame({'key': ['K0', 'K1', 'K2', 'K3', 'K4', 'K5'],
+        ...                      'A': ['A0', 'A1', 'A2', 'A3', 'A4', 'A5']},
+        ...                     columns=['key', 'A'])
+        >>> kdf2 = ks.DataFrame({'key': ['K0', 'K1', 'K2'],
+        ...                      'B': ['B0', 'B1', 'B2']},
+        ...                     columns=['key', 'B'])
+        >>> kdf1
+          key   A
+        0  K0  A0
+        1  K1  A1
+        2  K2  A2
+        3  K3  A3
+        4  K4  A4
+        5  K5  A5
+        >>> kdf2
+          key   B
+        0  K0  B0
+        1  K1  B1
+        2  K2  B2
+
+        Join DataFrames using their indexes.
+
+        >>> join_kdf = kdf1.join(kdf2, lsuffix='_caller', rsuffix='_other')
+        >>> join_kdf.sort_values(by=join_kdf.columns)
+          key_caller   A key_other     B
+        0         K0  A0        K0    B0
+        1         K1  A1        K1    B1
+        2         K2  A2        K2    B2
+        3         K3  A3      None  None
+        4         K4  A4      None  None
+        5         K5  A5      None  None
+
+        If we want to join using the key columns, we need to set key to be the index in both df and
+        other. The joined DataFrame will have key as its index.
+
+        >>> join_kdf = kdf1.set_index('key').join(kdf2.set_index('key'))
+        >>> join_kdf.sort_values(by=join_kdf.columns) # doctest: +NORMALIZE_WHITESPACE
+              A     B
+        key
+        K0   A0    B0
+        K1   A1    B1
+        K2   A2    B2
+        K3   A3  None
+        K4   A4  None
+        K5   A5  None
+
+        Another option to join using the key columns is to use the on parameter. DataFrame.join
+        always uses other’s index but we can use any column in df. This method preserves the
+        original DataFrame’s index in the result.
+
+        >>> join_kdf = kdf1.join(kdf2.set_index('key'), on='key')
+        >>> join_kdf.sort_values(by=join_kdf.columns)
+          key   A     B
+        0  K0  A0    B0
+        1  K1  A1    B1
+        2  K2  A2    B2
+        3  K3  A3  None
+        4  K4  A4  None
+        5  K5  A5  None
+        """
+        if on:
+            left = left.set_index('key')
+
+        left_keys = left._metadata.index_columns
+        right_keys = other._metadata.index_columns
+
+        if left_keys != right_keys:
+            raise ValueError(
+                'Index type and names should be same in the objects to join. '
+                'You passed different indices '
+                '{left_keys} and {right_keys}'.format(
+                    left_keys=left_keys, right_keys=right_keys))
+
+        left_table = left._sdf.alias('left_table')
```

Comment:
The following code looks a lot like the `merge()` method. Don't you think we could reuse that one?

Reply:
Yes,you remind me. It seems like `merg`e.Maybe i can use `merge` instead when https://github.com/databricks/koalas/issues/438 `merge` index issue fix done.Thanks for your kind guidance.